src/Simulation/Native/src/simulator/wavefunction.hpp

```diff
@@ -238,19 +290,55 @@ class Wavefunction
         rng_.seed(s);
     }
 
+    void reorderWFN(unsigned qubitLoc, unsigned newPos) const
+    {
+        //get id of qubit located at newPos and qubitLoc - getting index from the element
+        auto itr1 = std::find(qubitmap_.begin(), qubitmap_.end(), newPos);
+        auto newPosId = std::distance(qubitmap_.begin(), itr1);
+        auto itr2 = std::find(qubitmap_.begin(), qubitmap_.end(), qubitLoc);
+        auto qid = std::distance(qubitmap_.begin(), itr2);
+        //swap qubits in wfn between qubitLoc and newPos
+        if (newPos != qubitLoc)
+        {
+            for (unsigned i = 0; i < wfn_.size(); i++)
```

Comment:
the size of wavefunction store might be quite large, correct? is it guaranteed to be less than max unsigned? (otherwise the loop counter will wrap before reaching wfn_.size() and the loop won't terminate)

Reply:
Oh that is a great point! Let me look into this